infrastructures/kubernetes/src/main/java/org/eclipse/che/workspace/infrastructure/kubernetes/server/KubernetesServerExposer.java

```diff
@@ -290,6 +278,31 @@ private void exposeNonSecureServers(
     }
   }
 
+  private Service createService(
+      Map<String, ServerConfig> internalServers,
+      Map<String, ServerConfig> externalServers,
+      Map<String, ServicePort> unsecuredPorts,
+      Map<String, ServicePort> securedPorts) {
+    Map<String, ServerConfig> allInternalServers = new HashMap<>(internalServers);
+    // Adding external secure services, as they, in fact, internal also.
+    for (ServicePort servicePort : securedPorts.values()) {
+      Map<String, ServerConfig> securedExternalServers = match(externalServers, servicePort);
+      allInternalServers.putAll(securedExternalServers);
```

Comment:
Secure servers are logically internal, but they are not exposed through the service being created here but through JWT proxy. Why do you add them here to the list of the servers?